src/tests/test_utils/common_test_utils/include/common_test_utils/type_ranges.hpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <map>
+#include <vector>
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/core/type/element_type_traits.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+static ov::test::utils::InputGenerateData get_range_by_type(ov::element::Type temp_type, uint32_t max_range_limit = testing::internal::Random::kMaxRange) {
+    double min_start = 0 - (int32_t)round(max_range_limit / 2);
+
+    ov::test::utils::InputGenerateData inData(min_start, max_range_limit - 1);
+    auto apply_data = [&inData, &max_range_limit, &min_start](double new_range, double start_from)
+    {
+         if (new_range < max_range_limit) {
+            inData.start_from = start_from;
+            inData.range = (uint32_t)round(new_range);
+        } else {
+            inData.start_from = start_from > min_start ? start_from : min_start;
+            inData.range = max_range_limit - 1;
+        }
+    };
+
+#define CASE_T(X)                                                                             \
+    case X: {                                                                                 \
+        auto lowest = std::numeric_limits<ov::element_type_traits<X>::value_type>::lowest();  \
+        auto max = std::numeric_limits<ov::element_type_traits<X>::value_type>::max();        \
+        double tmp_range = static_cast<double>(max) - static_cast<double>(lowest);            \
+        apply_data(tmp_range, lowest);                                                        \
+        break;                                                                                \
+    }
+
+    switch (temp_type) {
+    case (ov::element::Type_t::undefined): {
+        inData.start_from = min_start;
+        inData.range = max_range_limit - 1;
+        break;
+    }
+    case (ov::element::Type_t::dynamic): {
+        inData.start_from = min_start;
+        inData.range = max_range_limit - 1;
+        break;
+    }
+    case (ov::element::Type_t::boolean): {
+        inData.start_from = 0;
+        inData.range = 2;
+        break;
+    }
+    case (ov::element::Type_t::bf16): {
+        ov::bfloat16 lowest_tmp = std::numeric_limits<ov::bfloat16>::lowest();
+        ov::bfloat16 max_tmp = std::numeric_limits<ov::bfloat16>::max();
+
+        double lowest = 0 - static_cast<double>(lowest_tmp.to_bits());
```

Comment:
to separate calculation of custom ov types limits to separated template function?

Reply:
moved to CASE_OV